pkg/operator/sync.go

```diff
@@ -186,11 +186,19 @@ func (optr *Operator) syncBaremetalControllers(config *OperatorConfig) error {
 		glog.Errorf("Unable to read Baremetal Provisioning config from CR %s.", baremetalProvisioningCR)
 		glog.Infof("Will try to read Baremetal Provisioning config from ConfigMap %s instead", baremetalConfigmap)
 	}
-	// Create a Secret needed for the Metal3 deployment
+	// Create Secrets needed for the Metal3 deployment
 	if err := createMariadbPasswordSecret(optr.kubeClient.CoreV1(), config); err != nil {
 		glog.Error("Not proceeding with Metal3 deployment. Failed to create Mariadb password.")
 		return err
 	}
+	if err := createIronicPasswordSecret(optr.kubeClient.CoreV1(), config, ironicSecretName, ironicUsername, "ironic"); err != nil {
+		glog.Error("Not proceeding with Metal3 deployment. Failed to create Ironic password.")
+		return err
+	}
+	if err := createIronicPasswordSecret(optr.kubeClient.CoreV1(), config, inspectorSecretName, inspectorUsername, "inspector"); err != nil {
+		glog.Error("Not proceeding with Metal3 deployment. Failed to create Ironic Inspector password.")
+		return err
+	}
 
```

Comment:
Nit: We should put all password creations into another method like createMetal3Secrets() in baremetal_pod.go. This method would return error when any one of the password creations fails. That way we are not referencing Ironic's specifics in sync.go.